frontends/PyCDE/src/pycde/system.py

```diff
@@ -101,11 +101,24 @@ def add_packaging_step(self, func: Callable):
 
   @property
   def hw_output_dir(self):
-    return self.output_directory / "hw"
+    ret = self.output_directory / "hw"
+    if not ret.exists():
+      ret.mkdir()
```

Comment:
So if the hw directory has existing file in it? what about cleaning it?
This function name doesn't make it clear that any action is expected from calling this function to get the path to the hw directory.

Reply:
I'm not big on deleting files, especially considering that there could be a compiled image in the build package.